test/ysh-runtime-errors.sh

```diff
@@ -491,6 +491,11 @@ test-var-decl() {
   _expr-error-case 'setvar x, y = 1, 2, 3'
 }
 
+test-const-decl() {
+  _error-case 'const x = {}; const x = {};'
+  _error-case 'const x; const x;'
+}
+
```

Comment:
Thanks, this is great!  I was just thinking about this bug, definitely needs to be fixed

Minor issue is that we don't have "allowed failures" for `test/ysh-runtime-errors.sh`, only for spec tests

So we could either 

- try to fix this bug and merge with this test
- Or add to  `spec/ysh-assign` so we can set `oils_allowed_failures:` .  Tests first are welcome as usual
- 

It's no problem to have  a bit of duplicate coverage in both test files, since this is just 2 lines

---

I haven't thought about / look at what the actual bug is, but we can chat about it

Reply:
Yeah, just having a poke about to see where it could be for now. I was thinking about option 1 but option 2 works well as well. Could be nice to have allowed failures in all test files.